share/spack/setup-env.sh

```diff
@@ -326,6 +326,16 @@ if [ "$_sp_shell" = bash ]; then
     export -f _spack_shell_wrapper
 fi
 
+# Identify and lock the python interpreter
+if [ -z "${SPACK_PYTHON}" ]; then
```

Comment:
This will fail when SPACK_PYTHON is not set if users use `set -e` in their shells (or in scripts where it's set).  Can you use the "idiom" (aka horribly unintuitive sh-ism) we use above, i.e.:

```
if [ -n "${_sp_initializing:-}" ]; then
...
```